### ISSUE TITLE
New @typescript-eslint v4 rules: no-shadow, no-redeclare

### ISSUE THREAD

**Issue by nihalgonsalves** [iamturns/eslint-config-airbnb-typescript#127]
`@typescript-eslint` v4 adds these new rules:

- https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md
- https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-redeclare.md

This is required since the base rules report false positives in TypeScript (e.g. `no-shadow` reports enums) with the v4 typescript-eslint release (due to breaking changes in the scope analysis engine).

- The `no-shadow` rule should be turned off and replaced with  `@typescript-eslint/no-shadow`
- The `no-redeclare` rule should be turned off and replaced with  `@typescript-eslint/no-redeclare`

**Comment by marcoreni** [comment]
Jumping in to report that the same is valid for `@typescript-eslint/no-redeclare`.

See v4 release notes here: https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0

**Comment by iamturns** [comment]
Fixed in #129 and release as v10.0.0 (thanks again @kripod )

**Comment by japalo** [comment]
I still had issues with this when using create react app. The issues went away after downgrading both this package and typescript. If this helps anyone out there getting stuck on this until every package is updated with the same workaround, it helped me to do the following:

First remove your node_modules. (`rm -rf node_modules`)
Then make sure your typescript is set to:
```
"typescript": "3.7.2"
```
Then, install previous version of this package:
```
yarn add -D eslint-config-airbnb-typescript@9.0.0 \
            eslint-plugin-jsx-a11y@^6.3.1 \
            eslint-plugin-react@^7.20.3 \
            eslint-plugin-react-hooks@^4.0.8 \
            @typescript-eslint/eslint-plugin@^3.6.1
```
**NOTE:** this is not a fix. This is a workaround while waiting for react-scripts to release their fix. They are aware of the problem. If you want to track it, here is the issue: https://github.com/facebook/create-react-app/issues/9515
